### PR TITLE
Expose QUnit version as property and display it in HTML Reporter

### PR DESCRIPTION
**Pull request by shivamdixit**
Screenshot:

qunit

Ref #554

### CODE REVIEW

**Comment by jzaefferer** [comment]
Showing the QUnit version in a separate line takes up too much visual space. Maybe put it in front of the user-agent, separated by a semicolon?

**Comment by shivamdixit** [comment]
@jzaefferer The length becomes too large if we display it in same line. Is it OK?

File: src/core.js [hunk 1]
```diff
@@ -184,6 +185,10 @@ config.modules.push( config.currentModule );
 
 	// Figure out if we're running the tests from a server or not
 	QUnit.isLocal = location.protocol === "file:";
+
+	// Expose the current QUnit version
+	// `version` initialized at top of scope
+	QUnit.version = version;
```
Comment (jzaefferer) [inline src/core.js]: Since `version` isn't used anywhere else, it can be inlined, that is, use `"@VERSION"` here.

File: reporter/html.js [hunk 2]
```diff
@@ -454,12 +455,23 @@ function storeFixture() {
...
+
 	if ( userAgent ) {
 		userAgent.innerHTML = "";
 		userAgent.appendChild( document.createTextNode( navigator.userAgent ) );
 	}
 }
 
+function appendVersion() {
+	var version = id( "qunit-version" ),
+		value = "@VERSION";
```
Comment (jzaefferer) [inline reporter/html.js]: Can this use `QUnit.version` instead?

**Comment by jzaefferer** [comment]
> The length becomes too large if we display it in same line. Is it OK?

What do you consider "too large"? It looks like it would fit fine on your screenshot. Vertical screenspace is more important than horizontal.

**Comment by shivamdixit** [comment]
Sure, I'll change it. (It looks little clumsy in my vertical monitor, for horizontal its perfectly fine)

**Comment by shivamdixit** [comment]
Refactored the code. And changed the display text. Screenshot:

qunit

File: reporter/html.js [hunk 3]
```diff
@@ -453,10 +453,14 @@ function storeFixture() {
...
 function appendUserAgent() {
-	var userAgent = id( "qunit-userAgent" );
+	var userAgent = id( "qunit-userAgent" ),
+		versionInfo = "QUnit " + QUnit.version;
+
 	if ( userAgent ) {
 		userAgent.innerHTML = "";
-		userAgent.appendChild( document.createTextNode( navigator.userAgent ) );
+		userAgent.appendChild(
+			document.createTextNode( navigator.userAgent + ": " + versionInfo )
```
Comment (jzaefferer) [inline reporter/html.js]: Let's replace the colon with a semicolon. Could also inline the `versionInfo` variable.

File: src/core.js [hunk 4]
```diff
@@ -184,6 +184,10 @@ config.modules.push( config.currentModule );
 
 	// Figure out if we're running the tests from a server or not
 	QUnit.isLocal = location.protocol === "file:";
+
+	// Expose the current QUnit version
+	// `version` initialized at top of scope
```
Comment (jzaefferer) [inline src/core.js]: This comment is now outdated and can be removed

File: src/qunit.css [hunk 5]
```diff
@@ -11,17 +11,17 @@
 
 /** Font Family and Sizes */
 
-#qunit-tests, #qunit-header, #qunit-banner, #qunit-testrunner-toolbar, #qunit-userAgent, #qunit-testresult {
+#qunit-tests, #qunit-header, #qunit-banner, #qunit-testrunner-toolbar, #qunit-version, #qunit-userAgent, #qunit-testresult {
```
Comment (jzaefferer) [inline src/qunit.css]: This looks like it can be reverted, since we're not creating a separate element anymore.

File: src/qunit.css [hunk 6]
```diff
@@ -68,6 +68,13 @@
 	overflow: hidden;
 }
 
+#qunit-version {
```
Comment (jzaefferer) [inline src/qunit.css]: Same here, and for the other changes above.

**Comment by jzaefferer** [comment]
Thanks, the update is an improvement. Just need to fix a few details.

**Comment by shivamdixit** [comment]
I'm not sure why the build is failing. Is the browserstack test are failing because of the UI changes that I made?

**Comment by jzaefferer** [comment]
Looks like the Travis build now passed. We run tests on browserstack, which sometimes fails with timeouts. Those are usually non-deterministic, which makes them difficult to debug and fix. Any persistent failures should be obvious enough though...

**Comment by jzaefferer** [comment]
This PR now looks good to land to me. Will let @leobalter take a look as well.

**Comment by leobalter** [comment]
Looks great, I have only one question: wouldn't it be better to show the QUnit version before the User Agent? It's a smaller text and would help locating every information on that line faster.

```js
"QUnit " + QUnit.version + " @ " + navigator.userAgent

**Comment by jzaefferer** [comment]
I had considered the same, since you brought it up: Yes, let's switch the order.

**Comment by shivamdixit** [comment]
Changed the order.
qunit

**Comment by jzaefferer** [comment]
I missed this in Leo's comment: Let's stick with the semicolon as the separator. "@" works fine when it refers to the version of something, like "npm install package@version", but here the QUnit version and userAgent are independent. (bikeshedded this with @arthurvr on IRC as well)

**Comment by shivamdixit** [comment]
@jzaefferer Okay. Fixed it.

**Comment by jzaefferer** [comment]
Thanks! Merged this, squashing commits into one. I also made a tiny edit, removing the space in front of the semicolon.

**Pull request closed without merging**